global.php

```diff
@@ -111,21 +111,36 @@
 // The user used our new quick theme changer
 if(isset($mybb->input['theme']) && verify_post_check($mybb->get_input('my_post_key'), true))
 {
-	$mybb->user['style'] = $mybb->get_input('theme');
-	// If user is logged in, update their theme selection with the new one
-	if($mybb->user['uid'])
+	// Set up user handler.
+	require_once MYBB_ROOT.'inc/datahandlers/user.php';
+	$userhandler = new UserDataHandler('update');
+
+	$user = array(
+		'uid'	=> $mybb->user['uid'],
```

Comment:
Need to pass `usergroup` and `displaygroup` here. In usercp.php's `do_options` action too.